test/translate_c.zig

```diff
@@ -21,9 +21,9 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
     , &[_][]const u8{
         \\pub export fn foo() void {
         \\    var a: c_int = undefined;
-        \\    var b: u8 = @as(u8, 123);
+        \\    var b: u8 = @intCast(u8, 123);
```

Comment:
Wouldn't we prefer `@as`?

Reply:
Yes, [but there seem to be some places `@intCast` is required](https://dev.azure.com/ziglang/zig/_build/results?buildId=4663&view=logs&j=28c53fb9-3e1b-538f-b6d2-6fc003711cae&t=b577613a-f97c-568b-2fe2-5ce83fbc18c3) and I'm not sure they can be done separately.